File: 2024/April/4-10 950 Reveal Cards in Increasing Order/CardReveal.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck);
        ArrayDeque<Integer> queue = new ArrayDeque<Integer>();

        for(int i = deck.length - 1; i > 0; i--) {
            queue.addFirst(deck[i]);
            queue.addFirst(queue.removeLast());
        }
        queue.addFirst(deck[0]);
        
        for(int i = 0; i < deck.length; i++) {
            deck[i] = queue.poll();
        }

        return deck;
    }
}

File: 2024/April/4-10 950 Reveal Cards in Increasing Order/CardReveal.py
from typing import List
from typing import Deque

class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        queue = Deque()
        deck.sort()
        deck.reverse()
        
        for num in deck:
            queue.appendleft(num)
            queue.appendleft(queue.pop())

        queue.append(queue.popleft())

        return list(queue)
        
